src/chat.rs

```diff
@@ -1783,24 +1846,36 @@ async fn prepare_msg_common(
     context: &Context,
     chat_id: ChatId,
     msg: &mut Message,
+    change_state_to: MessageState,
 ) -> Result<MsgId> {
-    msg.id = MsgId::new_unset();
     prepare_msg_blob(context, msg).await?;
     chat_id.unarchive(context).await?;
 
     let mut chat = Chat::load_from_db(context, chat_id).await?;
     ensure!(chat.can_send(context).await?, "cannot send to {}", chat_id);
 
-    // The OutPreparing state is set by dc_prepare_msg() before it
-    // calls this function and the message is left in the OutPreparing
-    // state.  Otherwise we got called by send_msg() and we change the
-    // state to OutPending.
-    if msg.state != MessageState::OutPreparing {
-        msg.state = MessageState::OutPending;
-    }
+    // check current MessageState for drafts (to keep msg_id) ...
+    let update_msg_id = if msg.state == MessageState::OutDraft {
+        msg.hidden = false;
```

Comment:
Why this line is needed?

Reply:
drafts are saved as "hidden" in `do_set_draft()` - this line makes sure, the reused draft record does not stay in the hidden state.